scripts/build/termux_step_finish_build.sh

```diff
@@ -1,7 +1,16 @@
 termux_step_finish_build() {
 	echo "termux - build of '$TERMUX_PKG_NAME' done"
 	test -t 1 && printf "\033]0;%s - DONE\007" "$TERMUX_PKG_NAME"
-	mkdir -p /data/data/.built-packages
-	echo "$TERMUX_PKG_FULLVERSION" > "/data/data/.built-packages/$TERMUX_PKG_NAME"
+
+	if [ -n "$TERMUX_ON_DEVICE_BUILD" ]; then
+		# For on-device builds cross-compiling is not supported so we can
+		# store information about built packages under $TERMUX_TOPDIR.
+		mkdir -p "$TERMUX_TOPDIR/.built-packages"
```

Comment:
How about introducing a new variable TERMUX_BUILD_DIRECTORY and setting it to $TERMUX_TOPDIR/.built-packages/ on device and /data/data/.built-packages/ on linux somewhere above, then simply modify the two lines above to use it? That way, you don't add more logic like this all over this pull.

Reply:
I can do this. Though `.built-package` is used only in few places and is not intended to be used in `build.sh` scripts.

Reply:
Done.